### PR TITLE
feat(GODT-1970): Soft limits

### PR DESCRIPTION
**Pull request by LBeernaertProton**
Add limit checks for mailbox count, messages per mailbox, UID and UIDValidity so that we don't run into issues related exceeding the maximum representable size for each of these data types (uint32).

### CODE REVIEW

File: limits/imap.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+	if nextUIDCount > i.maxUID || nextUIDCount < int(existingUID) {
+		return ErrMaxUIDReached
+	}
+
+	return nil
+}
+
+func (i IMAP) CheckUIDValidity(uid imap.UID) error {
+	if int(uid) == i.maxUIDValidity {
```
Comment (cuthix) [inline limits/imap.go:47]: and `>=` here ?

File: limits/imap.go [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+// IMAP contains configurable upper limits that can be enforced by the Gluon server.
+type IMAP struct {
+	maxMailboxCount           int
+	maxMessageCountPerMailbox int
+	maxUIDValidity            int
+	maxUID                    int
+}
+
+func (i IMAP) CheckMailBoxCount(mailboxCount int) error {
+	if mailboxCount == i.maxMailboxCount {
```
Comment (cuthix) [inline limits/imap.go:19]: why not using `>=` here?

Comment (LBeernaertProton) [reply limits/imap.go:19]: mailboxes are always create 1 at a time

File: limits/imap.go (hunk 1 above)
Comment (LBeernaertProton) [reply limits/imap.go:47]: mailboxes are always create 1 at a time

Comment (LBeernaertProton) [reply limits/imap.go]: But just in case, changed.

File: limits/imap.go (hunk 2 above)
Comment (LBeernaertProton) [reply limits/imap.go]: But just in case, changed.

**Review by cuthix — APPROVED** [review]

**Pull request merged**